packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart

```diff
@@ -0,0 +1,328 @@
+import 'dart_file_open_dialog_api.dart';
+import 'dart_shell_item_api.dart';
+
+/// Dart native implementation of FileSelectorAPI
+class DartFileSelectorAPI extends FileDialog {
+  /// We need the file to exist. This value defaults to `false`.
+  DartFileSelectorAPI(
+      [FileOpenDialogAPI? fileOpenDialogAPI, ShellItemAPI? shellItemAPI])
+      : super() {
+    _fileOpenDialogAPI = fileOpenDialogAPI ?? FileOpenDialogAPI();
+    _shellItemAPI = shellItemAPI ?? ShellItemAPI();
+  }
+
+  late FileOpenDialogAPI _fileOpenDialogAPI;
+  late ShellItemAPI _shellItemAPI;
+
+  /// Returns a directory path from user selection.
+  String? getDirectoryPath({
+    String? initialDirectory,
+    String? confirmButtonText,
+  }) {
+    fileMustExist = true;
+    final SelectionOptions selectionOptions = SelectionOptions(
+        allowMultiple: false, selectFolders: true, allowedTypes: <TypeGroup>[]);
+    return _getDirectory(
+        initialDirectory: initialDirectory,
+        confirmButtonText: confirmButtonText,
+        selectionOptions: selectionOptions);
+  }
+
+  /// Returns a full path, including file name and it's extension, from user selection.
+  String? getSavePath({
+    String? initialDirectory,
+    String? confirmButtonText,
+    String? suggestedFileName,
+    SelectionOptions? selectionOptions,
+  }) {
+    fileMustExist = false;
+    final SelectionOptions defaultSelectionOptions = SelectionOptions(
+        allowMultiple: false, selectFolders: true, allowedTypes: <TypeGroup>[]);
+    return _getDirectory(
+        initialDirectory: initialDirectory,
+        confirmButtonText: confirmButtonText,
+        suggestedFileName: suggestedFileName,
+        selectionOptions: selectionOptions ?? defaultSelectionOptions);
+  }
+
+  /// Returns a list of file paths.
+  List<String> getFiles(
+      {String? initialDirectory,
+      String? confirmButtonText,
+      required SelectionOptions selectionOptions}) {
+    fileMustExist = false;
+    int hResult = initializeComLibrary();
+    final IFileOpenDialog fileDialog = FileOpenDialog.createInstance();
+    using((Arena arena) {
+      final Pointer<Uint32> ptrOptions = arena<Uint32>();
+
+      hResult = getOptions(ptrOptions, fileDialog);
+      hResult = setDialogOptions(ptrOptions, selectionOptions, fileDialog);
+    });
+
+    hResult = setInitialDirectory(initialDirectory, fileDialog);
+    hResult = addFileFilters(selectionOptions, fileDialog);
+    hResult = addConfirmButtonLabel(confirmButtonText, fileDialog);
+    hResult = _fileOpenDialogAPI.show(hWndOwner, fileDialog);
+
+    return returnSelectedElements(hResult, selectionOptions, fileDialog);
+  }
+
+  /// Returns dialog options.
+  @visibleForTesting
+  int getOptions(Pointer<Uint32> ptrOptions, IFileOpenDialog dialog) {
+    final int hResult = _fileOpenDialogAPI.getOptions(ptrOptions, dialog);
+    _validateResult(hResult);
+
+    return hResult;
+  }
+
+  /// Returns the dialog option based on conditions.
```

Comment:
What sort of conditions? Looks like `FILEOPENDIALOGOPTIONS` and `SelectionOptions`. Specifically, document that options is a bitfield containing the union of options described in `FILEOPENDIALOGOPTIONS`.